Source/santad/SNTNotificationQueue.m

```diff
@@ -69,12 +69,30 @@ - (void)addEvent:(SNTStoredEvent *)event
   @synchronized(self.pendingNotifications) {
     [self.pendingNotifications addObject:d];
   }
-  [self flushQueue];
+  [self flushQueueWithReplies:(reply != nil)];
```

Comment:
As written today, `reply` is never `nil`, an empty block is provided. Will need to rework this flow. I think we probably should support nil block, but then the agent side will need some more smarts to first check for non-nil.

Reply:
Good point. I've made it so that now it defaults to being nil coming from SNTExecutionController but an empty reply block will be added for those cases in SNTNotificationQueue as passing a nil reply block over XPC is not allowed.